ironfish/src/account/accounts.ts

```diff
@@ -324,23 +324,27 @@ export class Accounts {
     }
   }
 
-  async updateHeadHash(headHash: Buffer | null): Promise<void> {
+  async updateHeadHashes(headHash: Buffer | null): Promise<void> {
     for (const account of this.accounts.values()) {
-      if (headHash) {
-        await this.db.saveHeadHash(account, headHash.toString('hex'))
-      } else {
-        await this.db.removeHeadHash(account)
-      }
+      await this.updateHeadHash(account, headHash)
     }
   }
 
+  async updateHeadHash(account: Account, headHash: Buffer | null): Promise<void> {
+    const hash = headHash ? headHash.toString('hex') : null
+
+    this.headHashes.set(account.id, hash)
```

Comment:
Funny enough, we were never updating the `Accounts` `this.headHash(es)`. Never needed it, I guess, but now that we're adding more logic around this, it's going to come in handy to keep up to date.